src/app/cli/src/graphql.ml

```diff
@@ -214,40 +214,51 @@ module Make (Commands : Coda_commands.Intf) = struct
         obj "Wallet" ~doc:"An account record according to the daemon"
           ~fields:(fun _ ->
             [ pubkey_field ~resolve:(fun _ account ->
-                  Stringable.public_key account.Account.Poly.public_key )
+                  (* Hack: Account.Poly.t is only parameterized over 'pk once
+                   * and so in order for delegate to optional we must also make
+                   * account public_key optional even though it's always Some.
+                   * In an attempt to avoid a large refactoring, and also avoid
+                   * making a new record, we'll deal with a value_exn here and
+                   * be sad. *)
+                  Stringable.public_key
+                    ( account.Account.Poly.public_key
+                    |> Option.value_exn ?here:None ?error:None ?message:None )
+              )
             ; field "balance"
                 ~typ:(non_null AnnotatedBalance.obj)
                 ~doc:"A balance of Coda as a stringified uint64"
                 ~args:Arg.[]
                 ~resolve:(fun _ account -> account.Account.Poly.balance)
-            ; field "nonce" ~typ:(non_null string)
+            ; field "nonce" ~typ:string
                 ~doc:
                   "Nonces are natural numbers that increase each transaction. \
                    Stringified uint32"
                 ~args:Arg.[]
                 ~resolve:(fun _ account ->
-                  Account.Nonce.to_string account.Account.Poly.nonce )
-            ; field "receiptChainHash" ~typ:(non_null string)
+                  Option.map ~f:Account.Nonce.to_string
+                    account.Account.Poly.nonce )
+            ; field "receiptChainHash" ~typ:string
                 ~doc:"Top hash of the receipt chain merkle-list"
                 ~args:Arg.[]
                 ~resolve:(fun _ account ->
-                  Receipt.Chain_hash.to_string
+                  Option.map ~f:Receipt.Chain_hash.to_string
                     account.Account.Poly.receipt_chain_hash )
-            ; field "delegate" ~typ:(non_null string)
+            ; field "delegate" ~typ:string
                 ~doc:
-                  "The public key to which you are delegating (including the \
-                   empty key!)"
+                  "The public key to which you are delegating. Your own key \
```

Comment:
I would be confused, reading this as a first-time user of this API. Maybe we can restate it as:

```
"The public key to which you are delegating. If you are not delegating to anybody, then this would return your public key."
```